app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt

```diff
@@ -158,9 +129,36 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
 
     private fun handleChooseFolder(folderId: Long, folderDisplayName: String) {
         selectedFolderId = folderId
+        selectedFolderDisplayName = folderDisplayName
         unreadFolder.summary = folderDisplayName
     }
 
+    private fun loadSavedState(savedInstanceState: Bundle?) {
+        if (savedInstanceState != null) {
+            if (savedInstanceState.containsKey(STATE_SELECTED_ACCOUNT_UUID)) {
+                val accountUuid = savedInstanceState.getString(STATE_SELECTED_ACCOUNT_UUID)
+                if (accountUuid != null) {
+                    handleChooseAccount(accountUuid)
+                }
+
+                if (savedInstanceState.containsKey(STATE_SELECTED_FOLDER_ID) &&
+                    savedInstanceState.containsKey(STATE_SELECTED_FOLDER_SUMMARY)) {
+                    val folderId = savedInstanceState.getLong(STATE_SELECTED_FOLDER_ID)
+                    val folderSummary = savedInstanceState.getString(STATE_SELECTED_FOLDER_SUMMARY)
+                    if (folderSummary != null) {
+                        handleChooseFolder(folderId, folderSummary)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun Bundle.putLongIfPresent(key: String, value: Long?) {
```

Comment:
Would this be the correct place to put the putLongIfPresent function? This method to extending a class is pretty new to me and I'm unsure of best practices.

Reply:
^ @cketti
Also, This is ready for re-review

Reply:
If in doubt check other files. I usually put those helper extension functions below regular instance methods. In this case right above the companion object.

Reply:
Sounds good. Thanks for you patience